File: android/app/src/main/java/tut/flightbookingsystem/model/Passenger.java
package tut.flightbookingsystem.model;

import android.os.Parcel;
import android.os.Parcelable;

import java.io.Serializable;

public class Passenger implements Serializable, Parcelable {
    public long id;
    public String name;
    public String firstnames;
    public String surname;
    public String id_number;
    public String date_of_birth;
    public String gender;
    public String food_and_drink;
    public Long drink_id;
    public Long food_id;
    public long booking_id;
    public long flight_seat_id;
    public MealDrink drink;
    public MealFood food;
    public FlightSeat flight_seat;
    public Booking booking;
    public String created_at;
    public String updated_at;

    public Passenger() {
    }

    public Passenger(final Parcel in) {
        name = in.readString();
    }

    @Override
    public int describeContents() {
        return 0;
    }

    @Override
    public void writeToParcel(final Parcel dest,
                              final int flags) {
        dest.writeLong(id);
        dest.writeString(name);
        dest.writeString(firstnames);
        dest.writeString(surname);
        dest.writeString(gender);
        dest.writeString(food_and_drink);
        dest.writeString(id_number);
        dest.writeString(date_of_birth);
        if (drink_id != null) {
            dest.writeLong(drink_id);
        }
        if (food_id != null) {
            dest.writeLong(food_id);
        }
        dest.writeLong(flight_seat_id);
        dest.writeLong(booking_id);
        dest.writeSerializable(drink);
        dest.writeSerializable(food);
        dest.writeSerializable(flight_seat);
        dest.writeSerializable(booking);
        dest.writeString(created_at);
        dest.writeString(updated_at);
    }

    public static final Creator<Passenger> CREATOR = new Creator<Passenger>() {
        @Override
        public Passenger createFromParcel(final Parcel in) {
            return new Passenger(in);
        }

        @Override
        public Passenger[] newArray(final int size) {
            return new Passenger[size];
        }
    };
}

File: web/app/Passenger.php
<?php

namespace App;

use Illuminate\Database\Eloquent\Model;

class Passenger extends Model
{
  protected $appends =[
                       'name',
                       'drink_id',
                       'food_id',
                       'food_and_drink'
                     ];
  protected $fillable = [
    'firstnames',
    'surname',
    'last_name',
    'id_number',
    'date_of_birth',
    'gender',
    'booking_id',
    'flight_seat_id'
  ];

  public function drink()
  {
    return $this->hasOne('App\MealDrink');
  }

  public function food()
  {
    return $this->hasOne('App\MealFood');
  }

  public function booking()
  {
    return $this->belongsTo('App\Booking');
  }

  public function flight_seat()
  {
    return $this->hasOne('App\FlightSeat', 'id', 'flight_seat_id');
  }


  public function getNameAttribute()
  {
    return $this->firstnames . ' ' . $this->surname;
  }

  public function getFoodAndDrinkAttribute()
  {
    return "hello";
    return is_null($this->meal)
    ? null
    : $this->food->food->name
    . ' with '
    . $this->drink->drink->name;
  }

  public function getDrinkIdAttribute()
  {
    if(is_null($this->drink));
    {
      return null;
    }

    return $this->drink->drink->id;
  }

  public function getFoodIdAttribute()
  {
    if(is_null($this->food));
    {
      return null;
    }

    return $this->food->food->id;
  }
}
